core/CMakeLists.txt

```diff
@@ -17,20 +7,23 @@ endif(APPLE)
 
 add_spt3g_library(core SHARED
 	src/G3EventBuilder.cxx src/G3Frame.cxx src/G3TimeStamp.cxx
-	src/G3Pipeline.cxx src/G3Writer.cxx src/G3Reader.cxx src/python.cxx
+	src/G3Pipeline.cxx src/G3Writer.cxx src/G3Reader.cxx
 	src/G3InfiniteSource.cxx src/G3Logging.cxx src/G3PrintfLogger.cxx
 	src/G3Data.cxx src/G3Vector.cxx src/G3Map.cxx src/G3Timestream.cxx
         src/G3Timesample.cxx
 	src/G3TriggeredBuilder.cxx src/G3MultiFileWriter.cxx src/dataio.cxx
 	src/crc32.c ${CORE_EXTRA_SRCS}
 	src/G3NetworkSender.cxx src/G3SyslogLogger.cxx
 	src/G3PipelineInfo.cxx src/G3Quat.cxx
-	src/int_storage.cxx
+	src/int_storage.cxx src/pybindings.cxx
 )
 
 # Link dependencies
 target_link_libraries(core spt3g)
 
+# add python bindings
+add_spt3g_module(core src/python.cxx)
```

Comment:
Everything should go into the regular library, and the module is just doing the import into python.  An alternative implementation could be to move all the `PYBINDINGS` things into python.cxx and drop the python dependency in the shared library altogether, but there are so many pieces of code that use python (and have python things integrated into their headers and methods) that this would be a whole lot of work.

Reply:
Thanks, makes sense.